src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+namespace Microsoft.Quantum.QsCompiler.Templates
+{
+    public static class CppInterop
```

Comment:
I know that I originally mentioned that encapsulating logic in a `static class` was a good option but thinking a bit more about it, creating sub-classes seems like a better idea.

Sub-classes can inherit from the Bond-generated classes and implement methods that are specific for C++ interop:
```csharp
    public class ArgumentCpp : Argument
    {
        public ArgumentCpp(Argument argument)
        {
            this.Name = argument.Name;
            this.Position = argument.Position;
            this.Type = argument.Type;
            this.ArrayType = argument.ArrayType;
            this.Values = argument.Values.ToList();
        }

        public string InteropType() =>
            this.Type switch
            {
                DataType.BoolType => "char",
                DataType.IntegerType => "int64_t",
                DataType.DoubleType => "double",
                DataType.PauliType => "char",
                DataType.RangeType => "InteropRange*",
                DataType.ResultType => "char",
                DataType.StringType => "const char*",
                DataType.ArrayType => "InteropArray *",
            };
    }
 
    public class EntryPointOperationCpp : EntryPointOperation
    {
        public List<ArgumentCpp> InteropArguments;
 
        public EntryPointOperationCpp(EntryPointOperation entryPointOperation)
        {
            this.Name = entryPointOperation.Name;
            this.Arguments = entryPointOperation.Arguments.ToList();
            this.InteropArguments = entryPointOperation.Arguments.Select(arg => new ArgumentCpp(arg)).ToList();
        }
    }
```
Apologies for the churn.

Reply:
You would need to update _src/QsCompiler/Compiler/Templates/QirDriverCppEx.cs_ to make the `entryPointOperation` member of type `EntryPointOperationCpp`.

Reply:
done